File: VitaCare_API/VitaCare_API/Endpoints/MedicalWorker/GetAllDoctors/DoctorsGetAllEndpoint.cs
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Identity.Data;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Data.Sql;
using Microsoft.EntityFrameworkCore;
using VitaCare_API.Data;
using VitaCare_API.Helpers;

namespace VitaCare_API.Endpoints.MedicalWorker.GetAllDoctors
{
    [Route("api/[controller]")]
    [ApiController]
    public class DoctorsGetAllEndpoint : MyBaseEndpoint<NoRequest, DoctorsGetAllResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;

        public DoctorsGetAllEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }

        [HttpGet]
        public override async Task<DoctorsGetAllResponse> Obradi([FromQuery]NoRequest request)
        {
            var data = await _applicationDbContext.Doctor.Select(x => new DoctorsGetAllResponseDoctors
            {
                doctorID = x.DoctorID,
                firstName = x.FirstName,
                lastName = x.LastName
            }).ToListAsync();


            return new DoctorsGetAllResponse
            {
                Doctors = data
            };
        }
    }
}

File: front-end/src/app/medical-worker/medical-worker-make-an-appointment/endpoints/doctors-getall.endpoint.ts
import {MyBaseEndpoint} from "../../../MyBaseEndpoint";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {MyConfig} from "../../../My-Config";
import {Injectable} from "@angular/core";

@Injectable({providedIn: 'root'})
export class DoctorsGetallEndpoint implements  MyBaseEndpoint<void, DoctorsGetAllResponse>{
    constructor(public httpClient:HttpClient) {

    }

    obradi(request: void): Observable<DoctorsGetAllResponse> {
        let url = MyConfig.server_address + '/api/DoctorsGetAllEndpoint';

        return this.httpClient.get<DoctorsGetAllResponse>(url);
    }
}


export interface DoctorsGetAllResponse {
    doctors: DoctorsGetAllResponseDoctors[]
}

export interface DoctorsGetAllResponseDoctors {
    doctorID: number;
    firstName: string;
    lastName: string;
}


